localstack-core/localstack/testing/testselection/matching.py

```diff
@@ -93,8 +93,8 @@ def service_tests(self, services: list[str]):
     def passthrough(self):
         return lambda t: [t] if self.matching_func(t) else []
 
-    def directory(self):
-        return lambda t: [get_directory(t)] if self.matching_func(t) else []
+    def directory(self, paths: list[str] = None):
+        return lambda t: (paths or [get_directory(t)]) if self.matching_func(t) else []
```

Comment:
nits:
- Would it make sense to explain the behavior of this function quickly in a docstring?
- I think it would be great if this new behavior would be tested in the unit tests of this module:
https://github.com/localstack/localstack/blob/master/tests/unit/testing/testselection/test_matching.py

Reply:
Agreed, I will write some tests and documentation for it.